src/kernel.ts

```diff
@@ -434,7 +477,14 @@ class Kernel {
   }
 
   /**
-   * Create a Kernel Message given input attributes.
+   * Get the current kernel url.
+   */
+  private get _kernelUrl(): string {
+    return utils.urlJoinEncode(this._baseUrl, KERNEL_SERVICE_URL, this._id);
```

Comment:
I think it makes sense to make it a regular member then. It's too much extra work, and it avoids allocations each time it's used.